csharp/src/Ice/OutgoingResponseFrame.cs

```diff
@@ -17,26 +16,21 @@ public sealed class OutgoingResponseFrame : OutgoingFrame
         /// <summary>The result type; see <see cref="ZeroC.Ice.ResultType"/>.</summary>
         public ResultType ResultType => Data[0][0] == 0 ? ResultType.Success : ResultType.Failure;
 
-        private static readonly ConcurrentDictionary<(Protocol Protocol, Encoding Encoding), OutgoingResponseFrame>
-            _cachedVoidReturnValueFrames =
-                new ConcurrentDictionary<(Protocol Protocol, Encoding Encoding), OutgoingResponseFrame>();
-
         /// <summary>Creates a new outgoing response frame with a void return value.</summary>
         /// <param name="current">The Current object for the corresponding incoming request.</param>
         /// <returns>A new OutgoingResponseFrame.</returns>
-        public static OutgoingResponseFrame WithVoidReturnValue(Current current) =>
-            _cachedVoidReturnValueFrames.GetOrAdd((current.Protocol, current.Encoding), key =>
-            {
-                var data = new List<ArraySegment<byte>>();
-                var ostr = new OutputStream(key.Protocol.GetEncoding(), data);
-                ostr.WriteByte((byte)ResultType.Success);
-                _ = ostr.WriteEmptyEncapsulation(key.Encoding);
-                return new OutgoingResponseFrame(key.Protocol,
-                                                 key.Encoding,
-                                                 data,
-                                                 new OutputStream.Position(0, 1),
-                                                 ostr.Tail);
-            });
+        public static OutgoingResponseFrame WithVoidReturnValue(Current current)
```

Comment:
Why did you remove the caching?

Reply:
Because an interceptor can write to the binary context of the cached frame